lib/amqp/basic.ex

```diff
@@ -346,9 +346,17 @@ defmodule AMQP.Basic do
 
     consumer_pid = consumer_pid || self()
 
-    receiver = ReceiverManager.register_handler(chan.pid, consumer_pid, :consume)
+    pid =
+      case chan.consumer_spec do
+        nil ->
+          %{pid: pid} = ReceiverManager.register_handler(chan.pid, consumer_pid, :consume)
```

Comment:
let's remove line 347 and change the second parameter to `consumer_pid || self()` here